pootle/apps/pootle_store/models.py

```diff
@@ -1972,7 +1971,7 @@ def update(self, overwrite=False, store=None, only_newer=False, user=None,
                                                        update_unitids, user,
                                                        revision=update_revision,
                                                        submission_type=submission_type)
-            self.file_mtime = disk_mtime
+            self.file_mtime = self.get_file_mtime() or self.file_mtime
```

Comment:
It looked to me `self.file_mtime` was some value retrieved from elsewhere, but now that I re-read along with your comment, I realize it's exactly the same value the store already has.

I believe this, although more verbose, expresses the intention better:
```python
disk_mtime = self.get_file_mtime():
if disk_mtime is not None:
    self.file_mtime = disk_mtime
```